src/parser/sparqlParser/SparqlQleverVisitor.cpp

```diff
@@ -539,6 +730,43 @@ std::string Visitor::visitTypesafe(Parser::DataBlockValueContext* ctx) {
   AD_FAIL()  // Should be unreachable.
 }
 
+// ____________________________________________________________________________________
+GraphPatternOperation::Minus Visitor::visitTypesafe(
+    Parser::MinusGraphPatternContext* ctx) {
+  return {visitTypesafe(ctx->groupGraphPattern())};
+}
+
+// ____________________________________________________________________________________
+using GraphPattern = ParsedQuery::GraphPattern;
+namespace {
+GraphPattern wrap(GraphPatternOperation op) {
+  auto pattern = GraphPattern();
+  pattern._children.emplace_back(std::move(op));
+  return pattern;
+}
+}  // namespace
+
+GraphPatternOperation Visitor::visitTypesafe(
+    Parser::GroupOrUnionGraphPatternContext* ctx) {
+  auto children =
+      visitVector<ParsedQuery::GraphPattern>(ctx->groupGraphPattern());
+  if (children.size() > 1) {
+    // https://en.cppreference.com/w/cpp/algorithm/accumulate
+    // a similar thing is done in QueryPlaner::uniteGraphPatterns
+    auto foldOp = [](GraphPatternOperation op1, GraphPattern op2) {
+      return GraphPatternOperation{
+          GraphPatternOperation::Union{wrap(std::move(op1)), std::move(op2)}};
+    };
+    return std::accumulate(std::next(children.begin(), 2), children.end(),
+                           GraphPatternOperation{GraphPatternOperation::Union{
+                               std::move(children[0]), std::move(children[1])}},
+                           foldOp);
+  } else {
+    return GraphPatternOperation{
```

Comment:
At least the name of the return type is redundant.

Reply:
The `GraphPatternOperation` constructor that constructs from one of the variant types is explicit.